connector.go

```diff
@@ -129,6 +129,24 @@ func (c *connector) Connect(ctx context.Context) (driver.Conn, error) {
 		mc.maxWriteSize = mc.maxAllowedPacket
 	}
 
+	// Additional handling for result set optional metadata
+	if mc.cfg.ResultSetMetadata != "" {
+		err = mc.exec("SET resultset_metadata=" + mc.cfg.ResultSetMetadata)
```

Comment:
`handleParams` sends all `SET` at once. It would be better to use that. See also my remark about the option name.